### PR TITLE
Implement ActiveSet configuration that allows gem users to determine …

### PR DESCRIPTION
**Pull request by riyengar8**
…if nils will be treated as largest or smallest values in a sort.

@fractaledmind - This is still a little brute-forcey, but I thought it would be most efficient to put it in front of you and Pull-Request my way to elegance. I'd appreciate your comments when you get a chance. I am in no rush for this (it's not a feature I need for any reason at the moment). It's really just an excuse for me to touch the project on a lab day.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,32 @@ For pagination, like filtering, we don't enforce any view-layer specifics. You s
...
 
+## Configuration
+
+### ActiveSet.configuration.on_asc_sort_nils_come
+
+Example usage in an initializer:
+
+```
+  ActiveSet.configure do |c|
+    c.on_asc_sort_nils_come = first
```
Comment (fractaledmind) [inline README.md]: missing the colon for the symbol here

File: lib/active_set/sorting/active_record_strategy.rb [hunk 2]
```diff
@@ -41,15 +41,24 @@ def executable_instructions
...
+      # ASC => [nil, -2, -1, 1, 2]
+      # DESC => [2, 1, -1, -2, nil]
       def order_operation_for(set_instruction)
         attribute_model = set_instruction.attribute_model
 
         arel_column = set_instruction.arel_column
         arel_direction = direction_operator(set_instruction.value)
-        nil_sorter = arel_column.send(arel_direction == :asc ? :eq : :not_eq, nil)
+
+        nil_sorter = if ActiveSet.configuration.on_asc_sort_nils_come == :last
```
Comment (fractaledmind) [inline lib/active_set/sorting/active_record_strategy.rb]: Given the conditional fork now, let's just move this into a method in this class:

```ruby
def nil_sorter_for(column, direction)
  nil_sorter_operator = if ActiveSet.configuration.on_asc_sort_nils_come == :last
                          direction == :asc ? :eq : :not_eq
                        else
                          direction == :asc ? :not_eq : :eq
                        end
  
  arel_column.send(nil_sorter_operator, nil)
end
```

I don't like the branching logic of that `nil_sorter_operator` variable, but I can't think of anything concise and equally as clear at the moment

File: spec/support/shared_examples_for_a_sorted_collection.rb [hunk 3]
```diff
@@ -6,9 +6,20 @@
       instructions.each do |(path, dir)|
         left_value = sort_value_for(path: path, object: left_result)
         right_value = sort_value_for(path: path, object: right_result)
-        sortedness = dir.to_s == 'asc' ? -1 : 1
 
-        expect([0, nil, sortedness]).to include(left_value <=> right_value)
+        if (dir.to_s == 'asc')
+          if (ActiveSet.configuration.on_asc_sort_nils_come == :last)
+            expect(right_value).to (be >= left_value).or be_nil
```
Comment (fractaledmind) [inline spec/support/shared_examples_for_a_sorted_collection.rb]: I did not know about `.or`. I like this refactor. Thank you.

File: README.md (hunk 1 above)
Comment (riyengar8) [reply README.md]: Thanks. Fixed in c0660303e00130282748da56c3300158752de35d

File: lib/active_set/sorting/active_record_strategy.rb (hunk 2 above)
Comment (riyengar8) [reply lib/active_set/sorting/active_record_strategy.rb]: I agree with you that there feels like there must be a purer way of doing this. See c6a8b86 for the refactoring you suggested (which is an improvement at least).

**Comment by fractaledmind** [comment]
@riyengar8: version 0.8.2 with this addition is now published on RubyGems. Thanks again!

**Pull request merged**